lnwallet/channel.go

```diff
@@ -3198,15 +3199,28 @@ func (lc *LightningChannel) validateCommitmentSanity(theirLogCounter,
 	// Calculate the commitment fee, and subtract it from the
 	// initiator's balance.
 	commitFee := feePerKw.FeeForWeight(commitWeight)
+	commitFeeMsat := lnwire.NewMSatFromSatoshis(commitFee)
+
+	// Before we apply the fee, we'll check to see if it's even payable
+	// using the initiator's balance. If the fee is greater than their
+	// current balance, then this commitment would consume their entire
+	// output, meaning the reservation would be violated.
```

Comment:
This is now inconsistent with the check done in `createCommitmentTx`. There we allow this to happen by consuming the whole output, here we don't. As argued there, we should probably allow this to handle the `update_fee` edge case.

Reply:
Hmm yeah you're correct, will adjust it